### PR TITLE
Remove `token_filter` in _analyze API

### LINKED ISSUE #20283: Remove `token_filter` in _analyze API
When you define an analyzer, we only support `filter` instead of `token_filter`. See https://www.elastic.co/guide/en/elasticsearch/reference/master/analysis-custom-analyzer.html

But the `_analyze` API still supports both forms. I suggest that we remove support for `token_filter` in `_analyze` API.

See also #17843.

### PR DESCRIPTION
**Pull request by johtani**
Remove the param and change docs

Closes #20283

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/rest/action/admin/indices/RestAnalyzeActionTests.java [hunk 1]
```diff
@@ -181,6 +181,21 @@ public void testDeprecatedParamException() throws Exception {
...
+            .field("tokenizer", "keyword")
+            .array("token_filter", "lowercase")
+            .endObject().bytes();
+
+        analyzeRequest = new AnalyzeRequest("for test");
+
+        try {
+            RestAnalyzeAction.buildFromContent(content, analyzeRequest, new ParseFieldMatcher(Settings.EMPTY));
+        } catch (Exception e) {
+            assertThat(e, instanceOf(IllegalArgumentException.class));
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/rest/action/admin/indices/RestAnalyzeActionTests.java]: Maybe:
```java
Exception e = expectThrows(IllegalArgumentException.class, () -> RestAnalyzeAction.buildFromContent(content, analyzeRequest, new ParseFieldMatcher(Settings.EMPTY)));
assertThat(e.getMessage(), startsWith("Unknown parameter [token_filter]"));
```

That way you don't have to assert the type of the exception.

File: docs/reference/migration/migrate_5_0/rest.asciidoc [hunk 2]
```diff
@@ -67,8 +67,8 @@ removed in Elasticsearch 6.0.0.
 
 ==== Analyze API changes
 
-The deprecated `filters`/`token_filters`/`char_filters` parameter has been
-renamed `filter`/`token_filter`/`char_filter`.
+The deprecated `filters`/`char_filters` parameter has been
+renamed `filter`/`char_filter` and `token_filters` parameter has been removed.
```
Comment (nik9000) [inline docs/reference/migration/migrate_5_0/rest.asciidoc]: What about:
```
The `filters` and `char_filters` parameters have been renamed `filter` and `char_filter`. The `token_filters` parameter has been removed. Use `filter` instead.
```
?

**Comment by nik9000** [comment]
I left some wording suggestions but otherwise LGTM.

**Comment by johtani** [comment]
@nik9000 Thanks for reviewing. Updated docs and refactoring tests using expectThrows

**Pull request merged**